core/capabilities/webapi/trigger.go

```diff
@@ -1,18 +1,267 @@
+		signerKey:         signerKey,
+		rateLimiter:       rateLimiter,
+		lggr:              lggr.Named("WorkflowConnectorHandler"),
+	}
+
+	return handler, nil
+}
+
+// https://gateway-us-1.chain.link/web-trigger
+//   {
+//     jsonrpc: "2.0",
+//     id: "...",
+//     method: "web-trigger",
+//     params: {
+//       signature: "...",
+//       body: {
+//         don_id: "workflow_123",
+//         payload: {
+//           trigger_id: "web-trigger@1.0.0",
+//           trigger_event_id: "action_1234567890",
+//           timestamp: 1234567890,
+//           topics: ["daily_price_update"],
+//           params: {
+//             bid: "101",
+//             ask: "102"
+//           }
+//         }
+//       }
+//     }
+//   }
+
+// from Web API Trigger Doc
+// trigger_id          - ID of the trigger corresponding to the capability ID
+// trigger_event_id    - uniquely identifies generated event (scoped to trigger_id and sender)
+// timestamp           - timestamp of the event (unix time), needs to be within certain freshness to be processed
+// topics            - [OPTIONAL] list of topics (strings) to be started by this event (affects all topics if empty)
+// workflow_owners   - [OPTIONAL] list of workflow owners allowed to receive this event (affects all workflows if empty)
+// params            - key-value pairs that will be used as trigger output in the workflow Engine (translated to values.Map)
+
+type TriggerRequestPayload struct {
+	TriggerId      string     `json:"trigger_id"`
+	TriggerEventId string     `json:"trigger_event_id"`
+	Timestamp      int64      `json:"timestamp"`
+	Topics         []string   `json:"topics"`
+	Params         values.Map `json:"params"`
+}
+
+func (h *triggerConnectorHandler) HandleGatewayMessage(ctx context.Context, gatewayID string, msg *api.Message) {
+	body := &msg.Body
+	sender := ethCommon.HexToAddress(body.Sender)
+	if !h.rateLimiter.Allow(body.Sender) {
+		h.lggr.Errorw("request rate-limited")
+		return
+	}
+	if !h.allowedSendersMap[sender.String()] {
+		h.lggr.Errorw("Unauthorized Sender")
+		return
+	}
+	h.lggr.Debugw("handling gateway request", "id", gatewayID, "method", body.Method, "sender", sender)
+	var payload TriggerRequestPayload
+	err := json.Unmarshal(body.Payload, &payload)
+	if err != nil {
+		return
+	}
+
+	switch body.Method {
+	case workflow.MethodWebAPITrigger:
+		h.lggr.Debugw("added MethodWebAPITrigger message", "payload", string(body.Payload))
+		// TODO: Is the staleness check supposed to be in the gateway?
+		currentTime := time.Now()
+		// TODO: check against h.config.MaxAllowedMessageAgeSec
+		if currentTime.Unix()-3000 > payload.Timestamp {
+			// TODO: fix up with error handling update in design doc
+			response := Response{Success: false}
+			h.sendResponse(ctx, gatewayID, body, response)
+		}
+
+		// is this right?
+		// Sri did wrappedPayload, err := values.WrapMap(log.Data), does that work in this case?
+		wrappedPayload, _ := values.WrapMap(payload)
```

Comment:
IIUC, we would need to populate payloads with topics and params

Reply:
The payload that is received here is the payload with
```
 payload: {
          trigger_id: "web-trigger@1.0.0",
          trigger_event_id: "action_1234567890",
          timestamp: 1234567890,
          topics: ["daily_price_update"],
          params: {
            bid: "101",
            ask: "102"
          }
        }
```
so isn't this correct?

I've added a comment asking about the conversion to a different format for the executor.

Reply:
Talked with JT, he agrees this is likely sufficient, at least for now.